lib/diagnostic.rb

```diff
@@ -7,37 +7,55 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    the_length
+    the_longest_string
+    the_longest_string = list.max_by(&:length)
+    the_length = the_longest_string.length # find length of longest string returned
+    return the_length
   end
+#Source: Slackoverflow. Tested in pry
```

Comment:
Hey Conor you wrote a lot more code than we need to do this in Ruby. Here's a quick way we can do it:

`list.max_by(&:length)`

Also remember that in Ruby it automatically returns the last line of the method so you don't need to use the `return` keyword.